src/test/py/bazel/query_test.py

```diff
@@ -38,6 +38,53 @@ def testSimpleQuery(self):
     self._AssertQueryOutput('deps(//foo:top-rule, 1)', '//foo:top-rule',
                             '//foo:dep-rule')
 
+  def testBuildFilesForExternalRepos_Simple(self):
+    self.ScratchFile('WORKSPACE', [
+      'load("//:deps.bzl", "repos")',
+      'repos()',
+    ])
+    self.ScratchFile('BUILD.bazel')
+    self.ScratchFile('deps.bzl', [
+      'def repos():',
+      '    native.new_local_repository(',
+      '        name = "io_bazel_rules_go",',
+      '        path = ".",',
+      """        build_file_content = "exports_files(glob(['*.go']))",""",
+      '    )',
+    ])
+    self._AssertQueryOutputContains('buildfiles(//external:io_bazel_rules_go)',
+                            '//external:WORKSPACE', '//:deps.bzl', '//:BUILD.bazel')
+
+  def testBuildFilesForExternalRepos_IndirectLoads(self):
```

Comment:
Can you please add a test for what I asked about in the other PR: a bzl file that calls a function inside the same file, which would cause duplicates if we were using the stack trace and not deduping?